src/main/java/org/jabref/gui/entryeditor/EntryEditor.java

```diff
@@ -54,63 +54,60 @@
 import org.slf4j.LoggerFactory;
 
 /**
- * GUI component that allows editing of the fields of a BibEntry (i.e. the
- * one that shows up, when you double click on an entry in the table)
+ * GUI component that allows editing of the fields of a BibEntry (i.e. the one that shows up, when you double click on
+ * an entry in the table)
  * <p>
  * It hosts the tabs (required, general, optional) and the buttons to the left.
  * <p>
- * EntryEditor also registers itself to the event bus, receiving
- * events whenever a field of the entry changes, enabling the text fields to
- * update themselves if the change is made from somewhere else.
+ * EntryEditor also registers itself to the event bus, receiving events whenever a field of the entry changes, enabling
+ * the text fields to update themselves if the change is made from somewhere else.
  */
 public class EntryEditor extends BorderPane {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(EntryEditor.class);
 
+    private final BasePanel panel;
     private final BibDatabaseContext databaseContext;
     private final CountingUndoManager undoManager;
-    private final BasePanel panel;
-    private Subscription typeSubscription;
-    private final List<EntryEditorTab> tabs;
+    private final PreferencesService preferencesService;
+    private final EntryEditorPreferences entryEditorPreferences;
     private final FileUpdateMonitor fileMonitor;
-    /**
-     * A reference to the entry this editor works on.
-     */
-    private BibEntry entry;
+    private final DialogService dialogService;
+    private final ExternalFilesEntryLinker fileLinker;
+    private final TaskExecutor taskExecutor;
+    private final StateManager stateManager;
+    private final List<EntryEditorTab> tabs;
+    private Subscription typeSubscription;
+    private BibEntry entry;  // A reference to the entry this editor works on.
     private SourceTab sourceTab;
-
     @FXML private TabPane tabbed;
     @FXML private Button typeChangeButton;
     @FXML private Button fetcherButton;
     @FXML private Label typeLabel;
 
-    private final EntryEditorPreferences preferences;
-    private final DialogService dialogService;
-    private final ExternalFilesEntryLinker fileLinker;
-    private final TaskExecutor taskExecutor;
-    private final StateManager stateManager;
-
-    public EntryEditor(BasePanel panel, EntryEditorPreferences preferences, FileUpdateMonitor fileMonitor, DialogService dialogService, ExternalFileTypes externalFileTypes, TaskExecutor taskExecutor, StateManager stateManager) {
+    public EntryEditor(BasePanel panel, PreferencesService preferencesService, FileUpdateMonitor fileMonitor,
+                       DialogService dialogService, ExternalFileTypes externalFileTypes, TaskExecutor taskExecutor,
```

Comment:
Theoretically it should be possible to Inject all arguments. Not sure about side effects

Reply:
It seems to work properly. Even more, I have been checking all parameters and they were same Globals.XXX that injects DefaultInjector.